runtime/runtime/src/verifier.rs

```diff
@@ -417,6 +430,30 @@ fn validate_add_key_action(
     Ok(())
 }
 
+fn truncate_string(s: &str, limit: usize) -> String {
```

Comment:
I'm surprised that something like this is not present in default Rust string library..

(or is it there, but we need a custom method for a reason ?)

Reply:
I am not surprised -- doing this properly is a hard and domain-specific: you probably don't want to truncate 🏳️‍🌈 to 🏳️ and for that you need to pull substantial amount of unicode in. And then you can truncate to the number of visual characters, or to the number of fixed-width symbols, or to the number of bytes.